ansible/roles/pyenv/tasks/main.yml

```diff
@@ -0,0 +1,59 @@
+---
+
+- name: pyenv depencencies present
+  yum:
+    name: ['gcc', 'zlib-devel', 'bzip2', 'bzip2-devel', 'readline-devel', 'sqlite', 'sqlite-devel',
+    'openssl-devel', 'tk-devel', 'libffi-devel']
+    state: present
+
+- name: pyenv root present
+  file:
+    path: '{{ pyenv_root }}'
+    state: directory
+    mode: '0777'
+
+- name: pyenv source cloned
+  tags:
+      - pyenv
+  git:
+    repo: https://github.com/pyenv/pyenv.git
+    dest: '{{ pyenv_root }}'
+    version: eb68ec9488f0df1f668e9272dd5bd8854edf1dff
+
+- name: pyenv root set
+  tags:
+    - pyenv
+  lineinfile:
+    path: '{{ user_home }}/.bash_profile'
+    state: present
+    regex: ^export PYENV_ROOT={{ pyenv_root }}
+    line: 'export PYENV_ROOT={{ pyenv_root }}'
+    insertafter: EOF
+
+- name: pyenv PATH set
+  tags:
+    - pyenv
+  lineinfile:
+    path: '{{ user_home }}/.bash_profile'
+    state: present
+    regex: ^export PATH={{ pyenv_root }}/bin
+    line: 'export PATH={{ pyenv_root }}/bin:$PATH'
+    insertafter: EOF
+
+- name: pyenv initiated
+  tags:
+    - pyenv
+  lineinfile:
+    path: '{{ user_home }}/.bash_profile'
+    state: present
+    regex: '^if command -v pyenv 1>/dev/null 2>&1; then eval ".*";fi$'
+    line: 'if command -v pyenv 1>/dev/null 2>&1; then eval "$(pyenv init -)";fi'
+    insertafter: EOF
+
+- name: pyenv virtualenv present
+  tags:
+    - pyenv
+  git:
+    repo: https://github.com/pyenv/pyenv-virtualenv.git
+    dest: '{{ pyenv_root }}/plugins/pyenv-virtualenv'
+    version: 294f64f76b6b7fbf1a22a4ebba7710faa75c21f7
```

Comment:
Make this an ansible variable and put in the defaults.yml file for the role.

Reply:
Done